bluemira/optimisation/problem.py

```diff
@@ -94,15 +95,18 @@ def objective(self, x: np.ndarray) -> float:
     def df_objective(self, x: np.ndarray) -> np.ndarray:
         """The gradient of the objective function at ``x``."""
 
-    def eq_constraints(self) -> List[ConstraintT]:
+    @staticmethod
+    def eq_constraints() -> List[ConstraintT]:
         """The equality constraints on the optimisation."""
         return []
 
-    def ineq_constraints(self) -> List[ConstraintT]:
+    @staticmethod
+    def ineq_constraints() -> List[ConstraintT]:
```

Comment:
Again with the staticmethods... When you subclass, won't you get warnings for using self thereafter?